packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -199,7 +199,9 @@ class PageController extends ScrollController {
       position._cachedPage = page.toDouble();
       return Future<void>.value();
     }
-
+    if (!position.hasContentDimensions) {
```

Comment:
I am not sure that returning in these methods is the right solution here. If we are trying to manipulate the position, but it is not ready yet, we shouldn't ignore it and let the user know.

Reply:
In the case that we have the `controller` and the page was not render. we can just update the `_cachedPage`, and the UI will be reflect when the page is ready to rendered. And here is the comment of the `_cachedPage`:

```Dart
  // When the viewport has a zero-size, the `page` can not
  // be retrieved by `getPageFromPixels`, so we need to cache the page
  // for use when resizing the viewport to non-zero next time.
  double? _cachedPage;
```